app/models/xml_submission.rb

```diff
@@ -0,0 +1,69 @@
+class XMLSubmission
+  attr_accessor :response, :data
+
+  def initialize(response: nil, data: nil)
+    @response = response
+    @data = data
+    populate_from_odk(@data) if @data.is_a? String
+    populate_from_j2me(@data) if @data.is_a? Hash
```

Comment:
I'm not sure about this heuristic. Didn't the old code have a check for an xmlns or something?